activestorage/active.py

```diff
@@ -17,6 +19,25 @@
 from activestorage.hdf2numcodec import decode_filters
 
 
+def return_storage_type(uri):
+    """
+    Extract the gateway-protocol to infer what type of storage
+    """
+    try:
+        resp = requests.head(uri)
+    except requests.exceptions.MissingSchema:  # eg local file
+        return
+    response = resp.headers
+
+    # https files on NGINX don't have "gateway-protocol" key
+    if "gateway-protocol" in response:
+        if response["gateway-protocol"] == "s3":
+            print("Gateway protocol indicates S3 storage.")
+            return "s3"
+    else:
+        return "https"
```

Comment:
@bnlawrence this is the bit I was on about on the call

Reply:
That's great. I think we'd need to put the response to that in the json going to reductionist, rather than in the URI itself.